test/admin/post.spec.js

```diff
@@ -0,0 +1,230 @@
+const supertest = require('supertest');
+const bcrypt = require('bcryptjs');
+const app = require('../../app');
+const userService = require('../../app/services/users');
+const sessionsManager = require('../../app/services/sessions_manager');
+const errorsCatalog = require('../../app/schemas/errors_catalog');
+const errors = require('../../app/errors');
+
+const { create: createUser } = require('../../test/factory/users');
+
+const request = supertest(app);
+
+const postUser = (endpoint, data, token) =>
+  request
+    .post(endpoint)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${token}`)
+    .send(data);
+
+const hashPassword = async password => {
+  const salt = await bcrypt.genSaltSync(10);
+  const hashedPass = await bcrypt.hashSync(password, salt);
+  return hashedPass;
+};
+
+let rawAdmin = {};
+const adminEndpoint = '/admin/users';
+
+describe('/admin/users [POST]', () => {
+  const newUser = {
+    email: 'upgradeable.user@wolox.com.ar',
+    password: 'hola1234',
+    name: 'Martin',
+    last_name: 'Acosta'
+  };
+  beforeEach(async () => {
+    let userFound = {};
+    const username = 'admin';
+    const userLogin = {
+      email: `${username}@wolox.com.ar`,
+      isAdmin: true
+    };
+    userLogin.password = await hashPassword('hola1234');
+    await createUser(userLogin);
+    userFound = await userService.emailExists(userLogin.email);
```

Comment:
Its not neccessary search the user that you created.
the function createUser return the raw that you insert in the test database;
You can do:
```.js
const createdUser =  await createUser(userLogin);

```